### PR TITLE
Add EditModuleCommand and relevant tests

### PR DESCRIPTION
**Pull request by drkkjh**
Hybrid testing done

### CODE REVIEW

**Comment by drkkjh** [comment]
#101

File: src/test/java/seedu/address/testutil/EditModuleDescriptorBuilder.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import seedu.address.logic.commands.module.EditModuleCommand.EditModuleDescriptor;
+import seedu.address.model.module.Module;
+import seedu.address.model.module.ModuleCode;
+import seedu.address.model.module.ModuleDescription;
+import seedu.address.model.module.ModuleName;
+import seedu.address.model.tag.Tag;
+
+/**
+ * A utility class to help with building EditPersonDescriptor objects.
```
Comment (bryanngzh) [inline src/test/java/seedu/address/testutil/EditModuleDescriptorBuilder.java:15]: Should this be EditModuleDescriptor instead?

Comment (drkkjh) [reply src/test/java/seedu/address/testutil/EditModuleDescriptorBuilder.java]: Fixed, thanks! 👍

File: src/main/java/seedu/address/logic/parser/EditModuleCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        return new EditModuleCommand(moduleCode, editModuleDescriptor);
+    }
+
+    /**
+     * Parses {@code Collection<String> tags} into a {@code Set<Tag>} if {@code tags} is non-empty.
+     * If {@code tags} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<Tag>} containing zero tags.
+     */
+    private Optional<Set<Tag>> parseTagsForEdit(Collection<String> tags) throws ParseException {
+        assert tags != null;
```
Comment (leongdl135) [inline src/main/java/seedu/address/logic/parser/EditModuleCommandParser.java:69]: Good use of asserts to check if tags are not null!

File: src/main/java/seedu/address/logic/commands/module/DeleteModuleCommand.java [hunk 3]
```diff
@@ -43,6 +41,7 @@ public CommandResult execute(Model model) throws CommandException {
         List<Module> moduleList = model.getFilteredModuleList();
         Module moduleToDelete;
         int positionInList = 0;
+        boolean isInModuleList = false;
```
Comment (leongdl135) [inline src/main/java/seedu/address/logic/commands/module/DeleteModuleCommand.java:44]: Good job by reducing the usage of global variables!

**Review by leongdl135 — APPROVED** [review]
LGTM!

**Pull request merged**